annnufan/assignment3/founder.cpp

```diff
@@ -0,0 +1,74 @@
+#include "dictionary.h"
+#include <set>
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <iostream>
+
+class founder {
+	dictionary dict;
+	std::vector<std::vector<char>> grid;
+	std::set<std::string> words_from_grid;
+	bool find_before;
+	int n, m;
+	std::vector<std::vector<int>> moves;
+
+	void go_from_cell(int i, int j, std::vector<std::vector<bool>> used, std::string word) {
+		word += grid[i][j];
+		used[i][j] = true;
+		if (!dict.is_prefix(word)) {
+			return;
+		}
+		if (dict.is_word(word)) {
+			words_from_grid.insert(word);
+		}
+		for (auto k : moves) {

```

Comment:
This version of `for` copies each move into `k`. Iterating on a reference can solve that. Also, if you do not want to change the `moves` array, you can declare `k` as const reference for safety:
```C++
for (const auto& k : moves) { ...
```

Also... Why the name `k`? `i` and `j` are indices, `n` and `m` are sizes... But what is `k`?
I think a more descriptive name can be used, e.g. `move`.
`for (const auto& move : moves)` is more readable in my opinion.